mk/docker.mk

```
@@ -20,12 +20,18 @@ DOCKER_DOCKERFILE ?= Dockerfile
 DOCKER_IMAGE_BASE ?= quay.io/$(USER)/myapp
 DOCKER_IMAGE_TAG ?= $(shell git rev-parse --short HEAD)
 DOCKER_IMAGE ?= $(DOCKER_IMAGE_BASE):$(DOCKER_IMAGE_TAG)
+DOCKER_LOGIN_USER ?= $(USER)
+DOCKER_REGISTRY ?= quay.io
 # DOCKER_OPTS
 # DOCKER_RUN_ARGS
 
+.PHONY: docker-login
+docker-login:
+	$(DOCKER) login -u "$(DOCKER_LOGIN_USER)" -p "$(DOCKER_LOGIN_TOKEN)" $(DOCKER_REGISTRY)
+
 .PHONY: docker-build
 docker-build:  ## Build image DOCKER_IMAGE from DOCKER_DOCKERFILE using the DOCKER_CONTEXT_DIR
-	$(DOCKER) build -t "$(DOCKER_IMAGE)" -f $(DOCKER_DOCKERFILE) $(DOCKER_CONTEXT_DIR)
+	$(DOCKER) build --no-cache -t "$(DOCKER_IMAGE)" -f $(DOCKER_DOCKERFILE) $(DOCKER_CONTEXT_DIR)
```

```suggestion
	$(DOCKER) build $(DOCKER_BUILD_OPTS) -t "$(DOCKER_IMAGE)" -f $(DOCKER_DOCKERFILE) $(DOCKER_CONTEXT_DIR)
```
I'd rather to use a variable, and set that when we need; in this particular scenario, docker will be using cached layers to build the container image except when we pass DOCKER_BUILD_OPTS="--no-cache":

`make docker-build DOCKER_BUILD_OPTS=--no-cache`